src/main.c

```diff
@@ -730,19 +733,66 @@ static h2o_iovec_t *build_http2_origin_frame(h2o_configurator_command_t *cmd, yo
+{
+    char *p = alloca(strlen(tls13_ciphers) + 1), *saveptr = NULL;
+    strcpy(p, tls13_ciphers);
+    struct {
+        const char *name;
+        ptls_cipher_suite_t *cipher;
+    } cipher_suites[] = {
+        { "TLS_AES_128_GCM_SHA256", &ptls_openssl_aes128gcmsha256},
+        { "TLS_AES_256_GCM_SHA384", &ptls_openssl_aes256gcmsha384},
+        { "TLS_CHACHA20_POLY1305_SHA256", &ptls_openssl_chacha20poly1305sha256 },
+    };
+    int seen_tls_aes_128_gcm_sha256 = 0;
+    H2O_VECTOR(ptls_cipher_suite_t *) ret = {};
+    p = strtok_r(p, ":", &saveptr);
+    while (1) {
+        if (p == NULL)
+            break;
+        int found  = 0;
+        for (int i = 0; i < (sizeof(cipher_suites) / sizeof(cipher_suites[0])); i++) {
+            if (!strcmp(p, cipher_suites[i].name)) {
+                ret.entries = h2o_mem_realloc(ret.entries, sizeof(ret.entries[0]) * (ret.size + 1));
+                ret.entries[ret.size++] = cipher_suites[i].cipher;
+                if (cipher_suites[i].cipher == &ptls_openssl_aes128gcmsha256)
+                    seen_tls_aes_128_gcm_sha256 = 1;
+                found = 1;
+                goto Next;
+            }
+        }
+Next:
+        if (!found) {
+            h2o_configurator_errprintf(cmd, node, "Unexpected cipher suite. Expected one of: `TLS_AES_128_GCM_SHA256`, `TLS_AES_256_GCM_SHA384` or `TLS_CHACHA20_POLY1305_SHA256`");
+            return NULL;
+        }
+        p = strtok_r(NULL, ":", &saveptr);
+    }
+    ret.entries = h2o_mem_realloc(ret.entries, sizeof(ret.entries[0])  * (ret.size + 1));
+    ret.entries[ret.size++] = NULL;
+
+    if (!seen_tls_aes_128_gcm_sha256) {
+        h2o_configurator_errprintf(cmd, node, "Warning: not enabling TLS_AES_128_GCM_SHA256 might reduce TLS1.3 interoperability, see RFC 8446 9.1");
+    }
+
+    return ret.entries;
+}
+
 static int listener_setup_ssl(h2o_configurator_command_t *cmd, h2o_configurator_context_t *ctx, yoml_t *listen_node,
                               yoml_t **ssl_node, yoml_t **cc_node, yoml_t **initcwnd_node, struct listener_config_t *listener,
                               int listener_is_new)
 {
     SSL_CTX *ssl_ctx = NULL;
     yoml_t **certificate_file, **key_file, **dh_file, **min_version, **max_version, **cipher_suite, **ocsp_update_cmd,
         **ocsp_update_interval_node, **ocsp_max_failures_node, **cipher_preference_node, **neverbleed_node,
-        **http2_origin_frame_node;
+        **http2_origin_frame_node, **cipher_suite_tls13;
     h2o_iovec_t *http2_origin_frame = NULL;
     long ssl_options = SSL_OP_ALL;
     uint64_t ocsp_update_interval = 4 * 60 * 60; /* defaults to 4 hours */
     unsigned ocsp_max_failures = 3;              /* defaults to 3; permit 3 failures before temporary disabling OCSP stapling */
     int use_neverbleed = 1, use_picotls = 1;     /* enabled by default */
+    ptls_cipher_suite_t **tls13_cipher_suites = NULL;
```

Comment:
The naming convention that we have here is:
* the variable that holds the converted value uses the same name as the config parameter
* the variable that holds the `yoml_t *` has the `_node` suffix.

Following that convention, I think we want to name this `cipher_suite_tls13` and `yoml_t *` above as `cipher_suite_tls13_node`.

Separately, I think it might make sense to set the default to `ptls_openssl_cipher_suites` here, and in `listener_setup_ssl_picotls` always use the supplied value.